### PR TITLE
Use color correct Skia in Flutter (part 2)

### PR DESCRIPTION
**Pull request by mattsarett**
*Color correct Android software backend
*Color correct mac desktop backend
*Respecting color space when converting frames to PNG
*Minor Vulkan change

@brianosman @liyuqian

### CODE REVIEW

**Comment by mattsarett** [comment]
*Color correct Android software backend
*Color correct mac desktop backend
*Respecting color space when converting frames to PNG
*Minor Vulkan change

@brianosman @liyuqian

File: content_handler/vulkan_rasterizer.cc [hunk 1]
```diff
@@ -127,7 +127,7 @@ VulkanRasterizer::VulkanSurfaceProducer::CreateSurface(uint32_t width,
   GrBackendRenderTargetDesc desc;
   desc.fWidth = width;
   desc.fHeight = height;
-  desc.fConfig = kSBGRA_8888_GrPixelConfig;
+  desc.fConfig = kSRGBA_8888_GrPixelConfig;
```
Comment (mattsarett) [inline content_handler/vulkan_rasterizer.cc]: Is there a particular reason for using BGRA?  If so, I can back this out.

File: shell/common/picture_serializer.cc [hunk 2]
```diff
@@ -19,8 +19,9 @@ bool PngPixelSerializer::onUseEncodedData(const void*, size_t) {
 SkData* PngPixelSerializer::onEncode(const SkPixmap& pixmap) {
   SkDynamicMemoryWStream stream;
 
-  bool encode_result = SkEncodeImage(
-      &stream, pixmap, SkEncodedImageFormat::kPNG, 80 /* quality */);
+  SkPngEncoder::Options options;
+  options.fUnpremulBehavior = SkTransferFunctionBehavior::kRespect;
+  bool encode_result = SkPngEncoder::Encode(&stream, pixmap, options);
```
Comment (mattsarett) [inline shell/common/picture_serializer.cc]: New png encode API supports embedding color space profiles and color correct unpremultiplication.  Note the that the "quality" param is always ignored for png (png is lossless).

File: shell/platform/android/android_surface_software.cc [hunk 3]
```diff
@@ -18,13 +18,17 @@ namespace shell {
 
 namespace {
 
-bool GetSkColorType(int32_t buffer_format, SkColorType* color_type) {
+bool GetSkColorTypeAndSkColorSpace(int32_t buffer_format,
+                                   SkColorType* color_type,
+                                   sk_sp<SkColorSpace>* color_space) {
   switch (buffer_format) {
     case WINDOW_FORMAT_RGB_565:
```
Comment (mattsarett) [inline shell/platform/android/android_surface_software.cc]: When do we think we'll hit this case?

**Comment by Hixie** [comment]
I presume this is blocked on landing https://github.com/flutter/engine/pull/3776.

**Comment by mattsarett** [comment]
Correct.  We could try to land them together - but given the difficulties of the other CL, I think keeping them separate makes sense.

**Comment by brianosman** [comment]
I think this can be landed now. #3826 (updated version of #3776) was merged and appears to have stuck?

**Comment by Hixie** [comment]
What is the state of this PR?

**Comment by brianosman** [comment]
It's on my list to land this, although I need to re-land part 1 (again).

**Comment by Hixie** [comment]
@brianosman I'm going to close this PR in the meantime, please reopen it (or a new one) when you're ready for review and landing.

**Pull request closed without merging**